templates/download/index.twig

```diff
@@ -27,116 +24,104 @@
-			</label>
-			<label class="btn btn-default" onclick="showOS('#mac')">
-				<input type="radio" name="options" id="mac-button"><span class="fab fa-apple fa-3x"></span><br>macOS
-			</label>
-		</div>
-	</div>
-</div>
+{% macro osbutton(os, icon, title) %}
+<label class="btn btn-default" onclick="showOS('#{{ os }}')">
+	<input type="radio" name="options" id="{{ os }}-button"><span class="fab {{ icon }} fa-3x"></span><br>{{ title }}
+</label>
+{% endmacro %}
 
-<div id="linux-div" class="text-center hidden container">
-	<h2>{% trans %}Install LMMS on Linux{% endtrans %}</h2>
-	<p>{% trans %}Click one of the buttons below to download LMMS for Linux.{% endtrans %}</p>
-	{% if linstable %}
+{% macro ospage(os, title, subtitle, stable, pre, nightly, instruction, trailer) %}
+<div id="{{ os }}-div" class="text-center hidden container">
+	<h2>{{ title }}</h2>
+	<p>{{ subtitle }}</p>
+	{% if stable %}
 		<h3>{% trans %}Stable Versions{% endtrans %}</h3>
-		{% for lin in linstable %}
-			{{ download.printrelbutton(lin) }}
+		{% for asset in stable %}
+			{{ download.printrelbutton(asset) }}
 		{% endfor %}
-		{{ download.releasenotes(linstable[0]['release']['body'], 'linstable') }}
-		<p>{% trans %}Run <code>chmod +x ~/Downloads/*.AppImage</code> to set the <code>.AppImage</code> executable before running{% endtrans %}</p>
+		{{ download.releasenotes(stable[0].description, "#{os}-stable") }}
+		{% if instruction %}<p>{{ instruction|raw }}</p>{% endif %}
 	{% endif %}
-
-	{% if linpre %}
+	{% if pre %}
 		<h3>
-		{% if 'alpha' in linpre[0].release.tag_name %}
+		{% if 'alpha' in pre[0].gitRef %}
 			{% trans %}Alpha Versions{% endtrans %}
 		{% else %}
 			{% trans %}Beta Versions{% endtrans %}
 		{% endif %}
 		</h3>
-		{% for lin in linpre %}
-			{{ download.printrelbutton(lin, 'btn-warning') }}
+		{% for asset in pre %}
+			{{ download.printrelbutton(asset, 'btn-warning') }}
 		{% endfor %}
-
-		{{ download.releasenotes(linpre[0]['release']['body'], 'linpre') }}
+		{{ download.releasenotes(pre[0].description, "#{os}-pre") }}
 	{% endif %}
-	<hr>
-	<a data-toggle="collapse" href="#collapse_lin_other">{% trans %}Other options{% endtrans %}<br><i class="fas fa-angle-down"></i></a>
-	<div id="collapse_lin_other" class="panel-collapse collapse">
-		{% include 'download/linux.twig' %}
-	</div>
-</div>
-
-<div id="windows-div" class="text-center hidden container">
-	<h2>{% trans %}Install LMMS on Windows{% endtrans %}</h2>
-	<p>{% trans %}Click one of the buttons below (either 32bit or 64bit) to download LMMS for Windows{% endtrans %}</p>
-	<h3>{% trans %}Stable Versions{% endtrans %}</h3>
-	{{ download.printrelbutton(winstable[0]) }}
-	{{ download.printrelbutton(winstable[1]) }}
-
-	{% if winstable and winstable[0] %}
-		{{ download.releasenotes(winstable[0]['release']['body'], 'winstable') }}
+	{% if nightly %}
+		<h3>{% trans %}Nightly Versions{% endtrans %}</h3>
+		{% for asset in nightly %}
+			{{ download.printrelbutton(asset, 'btn-secondary') }}
```

Comment:
The files are named rather vaguely `linux.zip`, etc with the installers inside the zip file.

I attempted to fix this in twig with the following code:

```twig
		{% if "/artifact/" in rel.downloadUrl %}
			download="lmms--{{ rel.releaseName }}-{{ rel.platformName |replace({" ": "-"})|lower }}.zip"
		{% endif %}	
			href="{{ rel.downloadUrl }}">
```

... but due to [strict same-origin rules (see upvoted comment)](https://stackoverflow.com/a/28364499/3196753) this isn't easily done.  Instead, we may be able to [write our own `redirect()`](https://stackoverflow.com/a/32427935/3196753) function to accommodate this.

Reply:
Issue opened https://github.com/LMMS/lmms.io/issues/363.